slither/evm/convert.py

```diff
@@ -0,0 +1,160 @@
+import logging
+import sha3
+import sys
+from slither.core.declarations import (Contract, Function)
+from slither.core.cfg.node import Node
+
+logger = logging.getLogger('ConvertToEVM')
+
+def get_evm_instructions(obj):
+    
+    assert isinstance(obj, (Function, Contract, Node))
+    
+    if "KEY_EVM_INS" not in obj.context:
+        
+        try:
+            # Avoiding the addition of evm_cfg_builder as permanent dependency
+            from evm_cfg_builder.cfg import CFG
+        except ImportError:
+            logger.error("To use evm printer, you need to install evm-cfg-builder from ToB")
+            logger.error("Documentation: https://github.com/crytic/evm_cfg_builder")
+            logger.error("Installation: pip install evm-cfg-builder")
+            sys.exit(-1)
+
+        slither = obj.slither
+
+        if isinstance(obj, Node):
+            contract = obj.function.contract
+        elif isinstance(obj, Function):
+            contract = obj.contract
+        else:
+            contract = obj
+
+        # Get contract runtime bytecode, srcmap and cfg
+        contract_bytecode_runtime = slither.crytic_compile.bytecode_runtime(contract.name)
+        contract_srcmap_runtime = slither.crytic_compile.srcmap_runtime(contract.name)
+        contract_cfg = CFG(contract_bytecode_runtime)
+
+        # Get contract init bytecode, srcmap and cfg
+        contract_bytecode_init= slither.crytic_compile.bytecode_init(contract.name)
+        contract_srcmap_init = slither.crytic_compile.srcmap_init(contract.name)
+        contract_cfg_init = CFG(contract_bytecode_init)
+
+        # Get evm instructions for contract
+        if isinstance(obj, Contract):
+            # Combine the instructions of constructor and the rest of the contract
+            obj.context["KEY_EVM_INS"] = contract_cfg_init.instructions + contract_cfg.instructions
+            
+        # Get evm instructions for function
+        elif isinstance(obj, Function):
+            
+            function = obj
+            
+            # CFG depends on function being constructor or not
+            if function.is_constructor:
```

Comment:
I am not sure to follow why the constructor is named dispatcher?

Reply:
The reason is that `_dispatcher` is the only function recognised by `evm-cfg-builder` in bytecode_init. I expected to find the Contract name (old style)  or `constructor ()` or corresponding function selector hash. But guess the `_dispatcher` serves the role of the constructor in init code, given that there are no other functions. Todo: We could rename it appropriately in `evm-cfg-builder` by detecting that init bytecode is being parsed.